src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { useTodosContext } from '../store';
+import { apiClient } from '../../api/todos';
+import { ErrorOption } from '../../enum/ErrorOption';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    setError,
+    removeTodo,
+    deletingTodoIds,
+    updateTodoTitle,
+    toggleTodoCondition,
+  } = useTodosContext();
+  const [isHovered, setIsHovered] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editingValue, setEditingValue] = useState(todo.title);
+
+  const isDeleting = deletingTodoIds.includes(todo.id);
+
+  const loaderCondition = isDeleting || isLoading || todo.id === 0;
+
+  const inputRef = useRef<null | HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  });
+
+  const handlerDeleteTodo = async (todoId: number) => {
+    setIsLoading(true);
+
+    try {
+      await apiClient.deleteTodo(todoId);
+      removeTodo(todoId);
+    } catch (error) {
+      setError(ErrorOption.DeleteTodoError);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handlerUpdateTodo = async () => {
+    const cachedTitle = todo.title;
+
+    try {
+      if (editingValue.trim() === '') {
```

Comment:
```suggestion
      if (editingValue.trim()) {
```

Reply:
That is not works for me, so i leave that how i writed (on pic i rid all title of todo and that result 😥)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/67290782/0aada3c4-9ca5-464a-9b71-dfd5150c84b7)

Reply:
> That is not works for me, so i leave that how i writed (on pic i rid all title of todo and that result 😥) ![image](https://private-user-images.githubusercontent.com/67290782/292889421-0aada3c4-9ca5-464a-9b71-dfd5150c84b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2MTEwNDYsIm5iZiI6MTcwMzYxMDc0NiwicGF0aCI6Ii82NzI5MDc4Mi8yOTI4ODk0MjEtMGFhZGEzYzQtOWNhNS00NjRhLTliNzEtZGZkNTE1MGM4NGI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI2VDE3MTIyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3Mjg5ZTZjNDdmYmM0NTQ2M2RhOGVlNTQ3NDczNDgyN2E2ZDAzMzAzZTBlYWRiMWYyNmIzZmFhYzFmMDQ2ZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bHbB_fsFPNpmddmCkOsOyWXJrnH52GSYUDjkBvzJDfI)

I answered this question above